src/game_state/game_state.h

```diff
@@ -27,6 +27,20 @@ void game_state_on_backspace(void);
 // user presses "enter" -> add current guess to word bank if able to
 void game_state_on_submit(void);
 
+void game_state_handle_key_press(char inputChar);
+
+/**
+ * GETTERS
+ */
+
+// TODO: think of a smoother way to return info to the UI
+
+char get_current_guess_char(int idx);
```

Comment:
This are pretty shaky, need to think of a better way to handle pulling data from game state into the UI while still having clear boundaries between the two